### PR TITLE
FIX: show more accurate precalculation fee on "Not enough balance." exception

### LINKED ISSUE #2404: Incorrect price calculation with coin control
Sending 1000$ will cost less than sending 100$. Banking allures :-)
The fee drops (actually) exactly from 410$ to 411$. AND
Until 410$: The fast & medium option shows the same price. Only by 411$ AND 412$, the medium price option is higher, than the fast. From 413$ to 1000$ the fast price option is as expected more expensive.

Also, price jumps between 0$, 5$, 10$.
Type: HD SegWit (BIP84 Benach32 Native) with 24 words imported from https://iancoleman.io/bip39/

I observed this in CHF and switched to $ to open the ticket.

Important: I have 4 Coins: $406.82 / $12.45 and Freezed: 2'081.52 / $11'466.48. I think this may show the error. (406+12 are close to the magic 411$).

YES! - This is hot: I unlocked the freezes. Then, the price drop moves up from 411$ to 13'950$. Same behavior there. The medium option for 13'950$ is higher than the fast prices.

I allways just changed the amount. No changes in inputs or outputs. Allow fee bump On/Off shows no impact.

I first beleaved, that diffrent wallet types result in different behavior. As the phenomenon is not observable with my watch-ony wallet, imported from CoboVault. And is not observale with Legacy (P2PKH), SegWit (P2SH), Legacy (BIP 44PSPKH), Multisig Vault. But there i have either not used coin control or zero or only one UTXO.

### PR DESCRIPTION
**Pull request by limpbrains**
When the user tries to send more funds than a wallet balance, BW still tries to precalculate fee to show something. 
To do this all transaction output values was replaced with dust value. In result, this tx always smaller than maximum possible option, because it has smaller number of inputs. 

Now BW will create tx with the same amount of outputs, but one of them will be MAX. It will allow showing more accurate fee estimation.

Fixes #2404

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/55y9qgth1j96g156czt0v20q1r

[download apk](http://filestorage.bluewallet.io:1488/1608818199.apk)

File: class/wallets/legacy-wallet.js [hunk 1]
```diff
@@ -276,16 +276,16 @@ export class LegacyWallet extends AbstractWallet {
     if (!changeAddress) throw new Error('No change address provided');
 
     let algo = coinSelectAccumulative;
-    if (targets.length === 1 && targets[0] && !targets[0].value) {
-      // we want to send MAX
+    // if targets has output without a value, we want send MAX to it
```
Comment (Overtorment) [inline class/wallets/legacy-wallet.js:279]: this change should apply to multisig wallet as well

Comment (Overtorment) [reply class/wallets/legacy-wallet.js:279]: in fact, this logic in multisig wallet is duplicated for some reason. maybe some refactoring is due

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: class/wallets/legacy-wallet.js (hunk 1 above)
Comment (limpbrains) [reply class/wallets/legacy-wallet.js:279]: refactored

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xvhbf46g5c0u63zb5krz9gphwg

[download apk](http://filestorage.bluewallet.io:1488/1608826175.apk)

File: class/wallets/multisig-hd-wallet.js [hunk 2]
```diff
@@ -809,22 +807,9 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
     if (targets.length === 0) throw new Error('No destination provided');
     if (this.howManySignaturesCanWeMake() === 0) skipSigning = true;
 
-    if (!changeAddress) throw new Error('No change address provided');
+    const { inputs, outputs, fee } = this.coinselect(utxos, targets, feeRate, changeAddress);
```
Comment (Overtorment) [inline class/wallets/multisig-hd-wallet.js:810]: bueno.

**Review by Overtorment — APPROVED** [review]

**Review by marcosrdz — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**